.github/workflows/pr.yml

```diff
@@ -14,49 +14,160 @@ on:
   pull_request:
     branches: [ main ]
 
+env:
+  GPG_MAIL: 'infrastructure-eng@newrelic.com'
+  OHAI_PFX_CERTIFICATE_BASE64: ${{ secrets.OHAI_PFX_CERTIFICATE_BASE64 }} # base64 encoded
+  OHAI_PFX_PASSPHRASE:  ${{ secrets.OHAI_PFX_PASSPHRASE }}
```

Comment:
perhaps we could move this to the step where they are being used to reduce the scope of the secrets.

Reply:
I guess we are in the same situation where external Pr will not have this secrets

Reply:
This PR not only adds the part of `CGO_NEABLED=0` but also refactors the pipeline to test everything that is tested at pre-release at PR time too.

The approach I took was to copy and paste the pre-release pipeline as it is to the PR one and remove the part when the artifacts are uploaded. With this information, I am trying to answer your questions:

> perhaps we could move this to the step where they are being used to reduce the scope of the secrets.

This is one of the things I did not want to move because I want to copy de Prerelease pipeline as close as possible. So the env variables are exactly in the same spot. I wanted to make the least amount of changes so the refactor does not go out of our hands. The repository needs a refactor from the ground up but this is neither the PR nor the time to do it.

> I guess we are in the same situation where external Pr will not have this secrets

You are right, when I did this PR I did not think about this. But it was like that before in the PR, e2e was failing. I know I increased the failure surface but I think this is something that we should tackle in other PR. Let me know if we are in the same spot here :)

Reply:
sure sounds good to break it in different PRs, perhaps just leave a ticket or TODO ?